src/drake/shell.clj

```diff
@@ -103,7 +103,7 @@
         proc (.exec (Runtime/getRuntime)
                     cmd-for-exec
                     ^"[Ljava.lang.String;" env
-                    (fs/file fs/*cwd*))]
+                    (fs/file (or exec-dir fs/*cwd*)))]
```

Comment:
Interestingly this causes problems. I'm doing a step with a simple ````pwd```` call, and the output actually changes when exec-dir is nil between using fs/*cwd* or ```(or exec-dir fs/*cwd*)```.

I put these prints in:

```clojure
        _ (println "exec dir" exec-dir)
        _ (println "cwd" fs/*cwd*)
        _ (println "or'd" (or exec-dir fs/*cwd*))
        _ (println "cwd path" (fs/file fs/*cwd*))
        _ (println "or'd path" (fs/file (or exec-dir fs/*cwd*)))
        dir (fs/file (or exec-dir fs/*cwd*))
        proc (.exec (Runtime/getRuntime)
                    cmd-for-exec
                    ^"[Ljava.lang.String;" env
                    dir)
```

And the last argument to exec will vary in the following two examples:

Using ```(fs/file fs/*cwd*)``` as ```dir```:
```
exec dir nil
cwd #<File /home/chen/code/drake/sandbox>
or'd #<File /home/chen/code/drake/sandbox>
cwd path #<File /home/chen/code/drake/sandbox>
or'd path #<File /home/chen/code/drake/sandbox>
/home/chen/code/drake/sandbox
```

Using ```(fs/file (or exec-dir fs/*cwd*))``` as ```dir```:
```
exec dir nil
cwd #<File /home/chen/code/drake/sandbox>
or'd #<File /home/chen/code/drake/sandbox>
cwd path #<File /home/chen/code/drake/sandbox>
or'd path #<File /home/chen/code/drake/sandbox>
/home/chen/code/drake/sandbox/inner
```

Which makes no sense to me.